{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Introduction\n",
    "\n",
    "This is a *book* about microbial metabolic models, their reconstructions and analysis at the strain and the community level. \n",
    "It is intended to give only some insight from the user's perspective and not a thorough background on each analysis presented. \n",
    "Yet, the basics will do be shown but mostly *when* to use a type of analysis, *what* can we learn from it, *how* to interpret their results and what are the assumptions made.\n",
    "\n",
    "The book contains numerous examples *as programs*, including implementations of many concepts. \n",
    "Each chapter is generated from a self-contained Jupyter Notebook. \n",
    "You can click on the \"download\" button at the top-right of the chapter, and then select \".ipynb\" to download the notebook for that chapter, and you'll be able to execute the examples yourself. \n",
    "Many of the examples are generated by code that is hidden (for readability) in the chapters you'll see here. \n",
    "You can show this code by clicking the \"Click to show\" labels adjacent to these cells.\n",
    "\n",
    "This *book* is open source, and the latest version will always be available online [here](https://uvm-plaid.github.io/programming-dp/notebooks/intro.html). \n",
    "The source code is available [on GitHub](https://github.com/hariszaf/notes-on-met-mod-analysis). \n",
    "If you would like to fix a typo, suggest an improvement, or report a bug, please open an issue on GitHub.\n",
    "\n",
    "The techniques described in this book have developed out of the study of *data privacy*. For our purposes, we will define data privacy this way:\n",
    "\n",
    "```{prf:definition} M-models\n",
    ":label: M-models\n",
    "Genome-scale metabolic models (**M-models**) provide for a metabolic description of genotype–phenotype relationship without accounting explicitly for synthesis of enzymes. \n",
    "M-models employ Boolean logic statements relating genes, proteins, and reactions, or the Gene–Protein–Reaction associations, or Gene-Protein-Reactions (GPRs).\n",
    "A reaction can only carry a non-zero flux if its GPR statement evaluates to True {cite}`palsson2015systems`.\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " integrated models of metabolism and expression (ME-Models) account\n",
    "explicitly for the genotype–phenotype relationship. Macromolecular expression\n",
    "is directly integrated with cellular metabolism {cite}`palsson2015systems`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "plaintext"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
